adbtuifm: init at 0.5.8

Daru-san opened the pull request:
## Description of changes
Add package adbtuifm from https://github.com/darkhz/adbtuifm
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kirillrdy commented on pkgs/by-name/ad/adbtuifm/package.nix:
```diff
@@ -0,0 +1,29 @@
+    android-tools
+  ];
+  meta = with lib; {
+    description = "adbtuifm is a TUI-based file manager for the Android Debug Bridge, to make transfers between the device and client easier.";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

`Not end with a period (or any punctuation for that matter).`

Daru-san replied:
I'll go and fix that issue

kirillrdy commented on pkgs/by-name/ad/adbtuifm/package.nix:
```diff
@@ -0,0 +1,29 @@
+  version = "v0.5.8";
+  src = fetchFromGitHub {
+    owner = "darkhz";
+    repo = pname;
```
use of pname considered anti pattern https://github.com/NixOS/nixpkgs/issues/277994

Daru-san replied:
Okay I see, I'll be happy to make that change, but I don't fully understand the differences therer= 
```suggestion
 repo = "adbtuifm"
```

kirillrdy replied:
the issue is that `pname` may be same ,but not directly related to say "repo"name, eg if someone forks a repo, package `pname` stays same ,but all other references need to be updated.

this may seem like a small nit pick, but for people looking after many packages is a real enough concern

Daru-san replied:
That makes sense actually, it is a really small thing but I do see how it can have a sizeable impact on maintaining packages

kirillrdy commented:
@Daru-san also, please split into two commits

`maintainers: add daru-san`
`adbtuifm: init at v0.5.8`

Aleksanaa commented on pkgs/by-name/ad/adbtuifm/package.nix:
```diff
@@ -0,0 +1,29 @@
+}:
+buildGoModule rec {
+  pname = "adbtuifm";
+  version = "v0.5.8";
```
```suggestion
  version = "0.5.8";
```

The version must start with a digit

https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

Daru-san committed:
maintainers: add daru-san

kirillrdy commented:
@Daru-san sorry one last thing, update commit message and PR title to match version ( drop v )

Daru-san committed:
adbtuifm: init at 0.5.8

kirillrdy commented on pkgs/by-name/ad/adbtuifm/package.nix:
```diff
@@ -0,0 +1,29 @@
+    android-tools
+  ];
+  meta = with lib; {
+    description = "adbtuifm is a TUI-based file manager for the Android Debug Bridge, to make transfers between the device and client easier";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

> Not start with the package name.
> 
>     More generally, it should not refer to the package name.
> 

> Be short, just one sentence.

Daru-san replied:
Okay, I can fix that. 

As a general question, for small changes like this for a PR on nixpkgs, would it be better to have multiple commits fixing different issues or rather to rebase every time something needs to be changed?

kirillrdy replied:
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

one commit per "unit"

Daru-san replied:
I see, thank you

kirillrdy commented on pkgs/by-name/ad/adbtuifm/package.nix:
```diff
@@ -0,0 +1,29 @@
+  };
+  vendorHash = "sha256-voVoowjM90OGWXF4REEevO8XEzT7azRYiDay4bnGBks=";
+  buildInputs = [
+    android-tools
```
this is not a buildInput as adbtuifm builds without android-tools

I assume it's runtime dependency ? you can either expect user to install it manually or use wrapProgram

https://nixos.org/manual/nixpkgs/unstable/#fun-wrapProgram

sorry for adhoc feedback, i should have had a popper look earlier

Daru-san replied:
Actually, after inspecting the [adb.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/adb.nix) module I came to realize that it's better if the user manually installs adb since it needs proper configuration to work

Daru-san committed:
adbtuifm: remove android-tools build input

The android-tools package requires additional configuration, which
cannot be provided in this package

Daru-san committed:
adbtuifm: shorten meta.description attribute

Changed the description to adhere to the meta guidelines

Daru-san commented:
Merged :smile:
